miasm/ir/symbexec.py

```diff
@@ -965,14 +965,15 @@ def modified(self, init_state=None, ids=True, mems=True):
                     continue
                 yield mem, value
 
-    def dump(self, ids=True, mems=True):
+    def dump(self,init_state_engine=None, ids=True, mems=True):
         """
         Display modififed variables
+        @init_state_engine: StateEngine instance
         @ids: display modified ids
         @mems: display modified memory
         """
 
-        for variable, value in self.modified(None, ids, mems):
+        for variable, value in self.modified(dict(init_state_engine if init_state_engine is not None else {}), ids, mems):
```

Comment:
Maybe the patch be better with:
```python
 def dump(self,init_state=None, ids=True, mems=True)
    ...
    if init_state is None:
        init_state ={}
    ...
    for variable, value in self.modified(init_state, ids, mems):
        ....
```

Is it ok for you ?

Reply:
Hmmm just note that modified doesnt take in a state instance as the name implies but a base dictionary as its commented in the source code (so a dict(state) )
`@init_state: a base dictionary linking variables to their initial values`

So I guess passing dict() of a StateEngine or correcting the modified func as well.

Reply:
Well, as the state is a kind of dictionary, I may keep the dictionary as argument if you are ok with this (the `.symbols` return a dictionary as well)

Reply:
All goodie then !